### PR TITLE
AA-257: Return useful errors and on-chain revert reasons to the RPC response message

### PR DESCRIPTION
**Pull request by forshtat**

### CODE REVIEW

File: tests/rip7560/test_send_failed.py [hunk 1]
```diff
@@ -36,3 +39,82 @@ def test_eth_send_no_code_wrong_nonce(w3):
...
     ret = tx.send()
     assert_rpc_error(ret, "nonce too high", RPCErrorCode.INVALID_INPUT)
+
+
+def test_eth_send_account_validation_reverts(wallet_contract_rules, tx_7560):
+    tx_7560.sender = wallet_contract_rules.address
+    tx_7560.nonce = hex(2)
+    tx_7560.signature = to_prefixed_hex("revert-msg")
+
+    response = tx_7560.send()
```
Comment (drortirosh) [inline tests/rip7560/test_send_failed.py:49]: testing the wrong function: 
normal transaction don't return revert string in "eth_send", but only in "eth_call" (and "eth_estimate")

These are the methods that need to be checked (but first, we need to implement them)

Comment (forshtat) [reply tests/rip7560/test_send_failed.py:49]: Well, I think there is a difference between legacy and 7560 transactions in that we do want to report the reason for transaction being invalid. This **does** happen with legacy transactions, if you send one with invalid nonce or signature you will get an exception right in the `eth_send`. 
On the other hand, `eth_call` should provide the "returned result" of the execution frame, if there was any. But also, why would there be such a thing? I don't see `eth_call` being useful in the context of RIP-7560.

File: tests/utils.py [hunk 2]
```diff
@@ -2,9 +2,11 @@
 import time
 from functools import cache
 
+from eth_utils import (
+    to_checksum_address
```
Comment (drortirosh) [inline tests/utils.py:6]: unused. how it passed lints?

File: tests/rip7560/test_send_failed.py [hunk 3]
```diff
@@ -36,3 +38,75 @@ def test_eth_send_no_code_wrong_nonce(w3):
...
+
+def test_eth_send_paymaster_validation_reverts(paymaster_contract_7560, tx_7560):
+    tx_7560.paymaster = paymaster_contract_7560.address
+    tx_7560.paymasterData = to_prefixed_hex("revert-msg")
+
+    response = tx_7560.send()
+
+    assert_rpc_error(
+        response,
+        "validation phase reverted in contract paymaster: on-chain revert message string",
```
Comment (drortirosh) [inline tests/rip7560/test_send_failed.py:60]: how can it return a "on-chain revert message string" ? this value is encoded inside Error(string), so someone did decode it

**Review by drortirosh — COMMENTED** [review]
what is actually missing is test with a revert on `error CustomError()`, that gets this data (even if only methodsig) back to the caller.
(e.g. as  error: { reason: "whatever", code: -32000, data: "0x12345" }

currently, I have a reason "Error: validation phase reverted in contract account", and no extra info

**Review by drortirosh — APPROVED** [review]

**Pull request merged**